### ISSUE TITLE
Add ZoneHVAC:FourPipeFanCoil with different capacity control methods to library

### ISSUE THREAD

**Issue by asparke2** [NREL/OpenStudio#1341]
From user: ZoneHVAC:FourPipeFanCoil – expose all of the EnergyPlus fields so that FCUs with VAV fans can be used with them (version 1.5.0 only allows for CAV fans no matter what Capacity Control Method is chosen).

Suggested approach:
Add several ZoneHVAC:FourPipeFanCoil objects to the library containing several of the valid combinations of Fan type and Capacity Control Method.  Double-check that there aren't any critical fields for this object hidden by the inspector policy.

IE:
1. CV Four Pipe Fan Coil: ConstantFanVariableFlow, Fan:ConstantVolume
2. Cycling Four Pipe Fan Coil: CyclingFan, Fan:OnOff
3. VAV Var Water Four Pipe Fan Coil: VariableFanVariableFlow, Fan:VariableVolume
4?? VAV Const Water Four Pipe Fan Coil: VariableFanConstantFlow, Fan:VariableVolume (two different control strategies for the coil flows possible with Fan:VariableVolume, not sure if both types need to be in library)

Field: Capacity Control Method
This input denotes how the unit’s output is controlled in order to meet zone heating or cooling requirement. The choices are ConstantFanVariableFlow, CyclingFan, VariableFanVariableFlow or VariableFanConstantFlow. For ConstantFanVariableFlow, the fan speed is held constant to produce a fixed air flow rate whenever the unit is scheduled on. The hot water or chilled flow rate is varied so that the unit output matches the zone heating or cooling requirement. For CyclingFan, the fan speed is chosen so that the unit capacity is greater than or equal to the heating / cooling load and the fan is cycled to match unit output with the load. For VariableFanVariableFlow both air and water flow rates are varied to match the load. For VariableFanConstantFlow, the water flow rate is at full flow and the fan speed varies to meet the load.

Field: Supply Air Fan Name
The name of a fan component that composes part of the fan coil unit. Note that the fan’s
maximum flow rate should be the same as the maximum airflow rate of the fan coil unit and
the type of fan object should correspond to the capacity control method. Namely, for
ConstantFanVariableFlow a Fan:OnOff or Fan:ConstantVolume should be used. For
CyclingFan, a Fan:OnOff should be used. And for VariableFanVariableFlow or
VariableFanConstantFlow a Fan:VariableVolume should be chosen. The fan’s inlet node
should be the same as the outdoor air mixer’s mixed air node.
The fan’s outlet node should be the same as the cooling coil’s air inlet node

**Comment by asparke2** [comment]
@kbenne also, while you're in there could you add a Cycling PTAC?  It should have a Fan:OnOff and have a Supply Air Fan Operating Mode Schedule that is always zero.

**Comment by asparke2** [comment]
Here is some ruby code to generate the cycling PTAC:

````
require 'openstudio'

#PTAC DX Clg HW Htg

model = OpenStudio::Model::Model.new

always_on = model.alwaysOnDiscreteSchedule

# Make an always-zero schedule for the PTAC fan
twenty_four_hrs = OpenStudio::Time.new(0,24,0,0)
ptac_fan_mode_sch = OpenStudio::Model::ScheduleRuleset.new(model)
ptac_fan_mode_sch.setName("Cycling PTAC Fan Mode Always Zero")
ptac_fan_mode_sch.defaultDaySchedule().setName("Cycling PTAC Fan Mode Sch Default")
ptac_fan_mode_sch.defaultDaySchedule().addValue(twenty_four_hrs,0)

fan = OpenStudio::Model::FanOnOff.new(model,always_on)
fan.setName("Cycling PTAC Fan")
fan.setPressureRise(250) #TODO units

htg_coil = OpenStudio::Model::CoilHeatingWater.new(model,always_on)
htg_coil.setName("Cycling PTAC HW Htg Coil")

clg_coil = OpenStudio::Model::CoilCoolingDXSingleSpeed.new(model)
clg_coil.setName("Cycling PTAC 1 Spd DX Clg Coil")

ptac = OpenStudio::Model::ZoneHVACPackagedTerminalAirConditioner.new(model,
                                                                    always_on, 
                                                                    fan,
                                                                    htg_coil,
                                                                    clg_coil)
ptac.setName("Cycling PTAC DX Clg HW Htg")
ptac.setSupplyAirFanOperatingModeSchedule(ptac_fan_mode_sch)

puts model
````

**Comment by MatthewSteen** [comment]
+1

**Comment by MatthewSteen** [comment]
I think this one can be closed with 5c48d07b9930bf19db8003531a68776257791e51 and a3b0867dfce8d1ac1ee790b28e10703d4bcaa0ee.

**Comment by evanweaver** [comment]
Fixed per SHAs above.
